awx/main/tasks/system.py

```diff
@@ -240,15 +240,10 @@ def apply_cluster_membership_policies():
         logger.debug('Cluster policy computation finished in {} seconds'.format(time.time() - started_compute))
 
 
-@task(queue='tower_broadcast_all')
+@task(queue='tower_settings_change')
 def clear_setting_cache(setting_keys):
-    orig_len = len(setting_keys)
-    for i in range(orig_len):
-        for dependent_key in settings_registry.get_dependent_settings(setting_keys[i]):
-            setting_keys.append(dependent_key)
-    cache_keys = set(setting_keys)
-    logger.debug('cache delete_many(%r)', cache_keys)
-    cache.delete_many(cache_keys)
+    # notify the service to clear the cache
+    send_pg_notify('tower_settings_change', setting_keys)
```

Comment:
It seems like 2 types of messages are going to be going into the "tower_settings_change" channel - one with the task data for `clear_setting_cache` and one with just a list of setting_keys. I'd prefer to only deal with the task. You could have `run_clear_cache.py` process the task data and then call this method directly (similar to the dispatcher but without using a thread pool).